backend/src/xfd_django/xfd_api/api_methods/user.py

```diff
@@ -486,6 +487,13 @@ def update_user_v2(user_id, user_data, current_user):
                 ),
             )
 
+        if "user_type" in requested_fields:
+            if updates["user_type"] in ["globalView", "globalAdmin", "regionalAdmin"]:
+                if not user.email.endswith("cisa.dhs.gov"):
```

Comment:
If we left endwith("cisa.dhs.gov") the domain could still be `terrible.domain.cisa.dhs.gov`. We would want an exact match. Also in `settings.py` add `ALLOWED_ADMIN_DOMAINS="cisa.dhs.gov"`
```suggestion
email_value = (user.email or "").strip().lower()
email_parts = email_value.split("@")
email_domain = email_parts[-1] if len(email_parts) == 2 else ""
allowed_admin_domains = settings.ALLOWED_ADMIN_DOMAINS  # list[str], i.e., ["cisa.dhs.gov"]

if email_domain not in allowed_admin_domains:
     LOGGER.warning(
        "User role elevation denied due to disallowed domain",
        extra={
            "user_id": getattr(user, "user_id", None),
            "email_domain": email_domain,
        },
    )
    raise HTTPException(
        status_code=403,
        detail="User not authorized for requested user type.",
```